introspection-engine/introspection-engine-tests/src/test_api.rs

```diff
@@ -235,6 +243,35 @@ impl TestApi {
         self.args.tags()
     }
 
+    pub fn datasource_block_string(&self) -> String {
+        let no_foreign_keys =
+            self.is_vitess() && self.preview_features().contains(PreviewFeature::ReferentialIntegrity);
+
+        let (url, relation_mode) = if no_foreign_keys {
+            (self.connection_string.to_string(), "\nrelationMode = \"prisma\"")
+        } else {
+            (r#"env(TEST_DATABASE_URL)"#.to_string(), "")
+        };
```

Comment:
I don't understand this conditional (how the url is related to the no_foreign_keys)

Reply:
(because we get the url from TEST_DATABASE_URL in tests)

Reply:
That conditional was there before afaik. Happy to try to remove it.